packages/crayons-core/src/components/icon/icon.tsx

```diff
@@ -11,41 +22,181 @@ import icons from './icon-assets/icons.json';
+  @Watch('lazy')
+  private async setSVGState(): Promise<void> {
+    const { name, visible, library, lazy, dataSvg, src } = this;
+
+    if (!lazy) this.eval_xOb = true;
+    else this.eval_xOb = visible;
+
+    if (!Build.isBrowser || !this.eval_xOb) {
+      return;
+    }
+    if (!name && dataSvg) {
+      this.svgHTML = dataSvg;
+    } else if (name) {
+      try {
+        let url = src;
+        if (!src) {
+          url = this.getUrl(name, library);
+        } else url = `${src}/${name}.svg`;
+        if (url !== undefined) {
+          const svgEl = await getSVGElement(url);
+          this.applySVGMutation(library, name, svgEl);
+          this.svgHTML = svgEl.outerHTML;
+        } else {
+          throw 'Icon-URL Not Valid.';
+        }
+      } catch (ex) {
+        console.info(
+          `Cannot load ${name}|${library} from CDN. Please check the url & provide a live path.`,
+          ex
+        );
+        this.loadFallbackImage();
+        return;
+      }
+    } else {
+      console.error(
+        "Please provide valid props either 'name' or 'data-svg'.Check the usage docs."
+      );
+      return;
+    }
+  }
+
+  private getUrl(icon: string, lib: string): string {
+    const library = getIconLibrary(lib);
+    if (icon && library) {
+      return library.resolver(icon);
+    } else {
+      throw 'Icon name/library not registered.';
+    }
+  }
+
+  private applySVGMutation(library: string, icon: string, svgEl: SVGElement) {
+    const iconlibrary = getIconLibrary(library);
+    if (iconlibrary && iconlibrary.mutator) {
+      iconlibrary.mutator(svgEl, icon);
+    }
+  }
+
+  /** Fetches the icon and redraws it. Used to handle library registrations. */
+  redraw() {
+    this.setSVGState();
+  }
+
+  async loadFallbackImage() {
+    this.svgHTML = await fetchIcon(this.getUrl('image', 'system'));
+    this.dataSvg = this.svgHTML;
+  }
+
+  connectedCallback(): void {
+    watchIcon(this);
+    this.lazy &&
+      waitUntilVisible(
+        this.intersectionObserver,
+        this.xRootMargin,
+        this.el,
+        () => {
+          this.visible = true;
+          this.setSVGState();
+        }
+      );
+  }
+
+  disconnectedCallback(): void {
+    unwatchIcon(this);
+    if (this.intersectionObserver) {
+      this.intersectionObserver.disconnect();
+      this.intersectionObserver = undefined;
+    }
   }
 
-  componentWillLoad() {
-    this.setSVGState(this.name);
+  async componentWillLoad(): Promise<void> {
+    this.setSVGState();
   }
 
   render() {
-    const style = {
-      '--icon-color': this.color,
-    };
+    const style = {};
+    const accessibilityProps = {};
 
-    if (this.size !== undefined) {
-      style['--icon-size'] = `${this.size}px`;
+    const hasLabel = typeof this.label === 'string' && this.label.length > 0;
+    if (hasLabel) {
+      accessibilityProps['role'] = 'img';
+      accessibilityProps['aria-label'] = this.label;
+      accessibilityProps['aria-hidden'] = true;
```

Comment:
Should we not set the aria-hidden to `true` by default for icons. Here if the label is not passed, aria-hidden attribute will not be set.

Reply:
Fixed